crates/deltalake-core/src/operations/delete.rs

```diff
@@ -249,6 +249,19 @@ async fn execute(
 
     metrics.execution_time_ms = Instant::now().duration_since(exec_start).as_micros();
 
+    let mut app_metadata = match app_metadata {
+        Some(meta) => meta,
+        None => HashMap::new(),
+    };
+
+    app_metadata.insert("readVersion".to_owned(), snapshot.version().into());
+
+    let delete_metrics = serde_json::to_value(metrics.clone());
+
+    if let Ok(map) = delete_metrics {
+        app_metadata.insert("operationMetrics".to_owned(), map);
+    }
+
```

Comment:
I think we can avoid cloning here.

```suggestion
    app_metadata.insert("readVersion".to_owned(), snapshot.version().into());

    if let Ok(map) = serde_json::to_value(&metrics) {
        app_metadata.insert("operationMetrics".to_owned(), map);
    }

```